cur-lib/cur/stdlib/list.rkt

```diff
@@ -31,3 +31,63 @@
      z]
     [(cons (A : Type) (a : A) (rest : (List A)))
      (s (recur rest))]))
+
+(require cur/stdlib/tactics/base
+         cur/stdlib/tactics/standard)
+
+;; XXX Error messages are really bad, I made lots of mistakes and it
+;; was very painful to find them.
+
+;; XXX Move to different file
+(data ex : (forall (A : Type) (P : (-> A Type)) Type)
+      (ex-intro :
+                (forall
+                 ;; XXX I forgot to include these. It is annoying that
+                 ;; I have to
+                 (A : Type) (P : (-> A Type))
+                 (a : A)
+                 (-> (P a)
+                     (ex A P)))))
+
+(define-syntax-rule (exists (x : T) P)
+  (ex T (λ (x : T) P)))
+
+;; XXX Make a (try x) tactic which does (eapply (ex-intro A P x _))
+;; reading A and P from the current goal.
```

Comment:
The proof-state supports a goals list, spawning a new goal should just be adding to the list. I haven't thought enough about how to properly handle depending on other goals.